lib/util/fs.js

```diff
@@ -0,0 +1,19 @@
+const fs = require('fs');
+const { promisify } = require('util');
+const mkdirp = require('mkdirp');
+
+/**
+ * Ensures the given path exists.
+ *  - If the path already exist, it's fine - it does nothing.
+ *  - If the path doesn't exist, it will create it.
+ *
+ * @param {string} path
+ * @returns {Promise}
+ */
+function ensureDirectoryExists(dir) {
+  return promisify(fs.stat)(dir).catch(() => mkdirp(dir));
```

Comment:
Can you extract `promisify(fs.stat)` outside of the function to make sure it is only invoked once?

Reply:
Done.